Wrappers/Python/ccpi/io/TIFF.py

```diff
@@ -0,0 +1,84 @@
+    
+    def __init__(self,
+                 **kwargs):
+        
+        self.data_container = kwargs.get('data_container', None)
+        self.file_name = kwargs.get('file_name', None)
+        counter_offset = kwargs.get('counter_offset', 0)
+        
+        if ((self.data_container is not None) and (self.file_name is not None)):
+            self.set_up(data_container = self.data_container,
+                        file_name = self.file_name, 
+                        counter_offset=counter_offset)
+        
+    def set_up(self,
+               data_container = None,
+               file_name = None,
+               counter_offset = -1):
+        
+        self.data_container = data_container
+        self.file_name = os.path.splitext(os.path.basename(file_name))[0]
+        self.dir_name = os.path.dirname(file_name)
+        self.counter_offset = counter_offset
+        
+        if not ((isinstance(self.data_container, ImageData)) or 
+                (isinstance(self.data_container, AcquisitionData))):
+            raise Exception('Writer supports only following data types:\n' +
+                            ' - ImageData\n - AcquisitionData')
+
+    def write_file(self):
+        '''alias of write'''
+        return self.write()
+    
+    def write(self):
+        if not os.path.isdir(self.dir_name):
+            os.mkdir(self.dir_name)
+
+        ndim = len(self.data_container.shape)
+        if ndim == 2:
+            # save single slice
+            
+            if self.counter_offset >= 0:
+                fname = "{}_idx_{:04d}.tiff".format(os.path.join(self.dir_name, self.file_name), self.counter_offset)
+            else:
+                fname = "{}.tiff".format(os.path.join(self.dir_name, self.file_name))
+            with open(fname, 'wb') as f:
+                Image.fromarray(self.data_container.as_array()).save(f, 'tiff')
+        elif ndim == 3:
+            for sliceno in range(self.data_container.shape[0]):
+                # save single slice
+                # pattern = self.file_name.split('.')
+                dimension = self.data_container.dimension_labels[0]
+                fname = "{}_idx_{:04d}.tiff".format(
+                    os.path.join(self.dir_name, self.file_name),
+                    sliceno + self.counter_offset)
+                with open(fname, 'wb') as f:
+                    Image.fromarray(self.data_container.as_array()[sliceno]).save(f, 'tiff')
+        elif ndim == 4:
+            for sliceno1 in range(self.data_container.shape[0]):
+                # save single slice
+                # pattern = self.file_name.split('.')
+                dimension = [ self.data_container.dimension_labels[0] ]
+                for sliceno2 in range(self.data_container.shape[1]):
+                    idx = self.data_container.shape[0] * sliceno2 + sliceno1 + self.counter_offset
+                    fname = "{}_{}_{}_idx_{}.tiff".format(os.path.join(self.dir_name, self.file_name), 
```

Comment:
index should have `{:04d}`

Reply:
Let's use a file name containing the shape of the file we are saving and 2 counters, the first for the 4th dimensions and the 2nd for the 3rd. The s1st and 2nd indices do not matter as data is read 2D by 2D.

This would allow to read a TIFF stack back in a `DataContainer` without geometry by the `TIFFReader`. Additionally the reader should provide a method that accepts a geometry and the reader could read in a TIFF stack as `ImageData` or `AcquisitionData`

```python

filename_shape_5x120x256x240_05_001.tiff

reader = TIFFReader()
reader.set_filename("filename_*_*.tiff")
# returns a DataContainer without geometry
dc = reader.get_output()
# then to read in as ImageData or AcquisitionData
ad = ag.allocate(None)
ad.fill(dc)
# or
acqd = AcquisitionData(dc.as_array(), deep_copy=False, geometry=ag)

# returns an ImageData with provided geometry
reader.read_as_ImageData(ig)
imgd = reader.get_output()
# or 
# returns an AcquisitionData with provided geometry
reader.read_as_AcquisitionData(ag)
acqd = reader.get_output()
```